pkg/common/nodeutil/node.go

```diff
@@ -0,0 +1,15 @@
+package nodeutil
+
+import (
+	"github.com/spiffe/spire/proto/spire/common"
+)
+
+// IsAgentBanned determines if a given attested node is banned or not.
+// An agent is considered as "banned" if its serial numbers (current and new)
+// are set to empty.
+func IsAgentBanned(node common.AttestedNode) bool {
```

Comment:
Maybe having `*common.AttestedNode` is better?

Reply:
I prefer a bool, because we can use it to assign a value to our proto responses

Reply:
I thought about that too. I wasn't clear for me which would be the right thing to do here. 

What discouraged me from using a pointer as argument was that the function would have to return an error in case the pointer is nil. And that could make the function a bit more "uncomfortable" to use.

Also I noticed that, in general, the possible callers of this function have already checked the node for being nil before calling IsAgentBanned... so an extra nil check could become redundant?

Reply:
That sounds fair.

Reply:
I'd probably lean towards `*common.AttestedNode`? That is more natural for proto messages. I'd probably just have this function assume the node is not `nil`, since this is really just syntactic sugar around "how to determine banned given an attested node`. Callers who haven't already checked for non-nil are going to just call this like `IsAgentBanned(*node)` and panic anyway.

Reply:
@marcosy the `IsAgentBanned` func has been implemented here: https://github.com/spiffe/spire/pull/1685.
That's now merged in master.

Reply:
Great! I'll update this PR